packages/expo-modules-autolinking/src/ExpoModuleConfig.ts

```diff
@@ -0,0 +1,36 @@
+import { RawExpoModuleConfig, SupportedPlatform } from './types';
+
+/**
+ * A class that wraps the raw config (`expo-module.json` or `unimodule.json`).
+ */
+export class ExpoModuleConfig {
+  constructor(readonly rawConfig: RawExpoModuleConfig) {}
+
+  /**
+   * Whether the module supports given platform.
+   */
+  supportsPlatform(platform: SupportedPlatform): boolean {
+    return this.rawConfig.platforms?.includes(platform) ?? false;
+  }
+
+  /**
+   * Returns a list of names of Swift native modules classes to put to the generated modules provider file.
+   */
+  iosModulesClassNames() {
+    return this.rawConfig.ios?.modulesClassNames ?? [];
+  }
+
+  /**
+   * Returns serializable raw config.
+   */
+  toJSON(): RawExpoModuleConfig {
+    return this.rawConfig;
+  }
+}
+
+/**
+ * Reads the config at given path and returns the config wrapped by `ExpoModuleConfig` class.
+ */
+export function requireAndResolveExpoModuleConfig(path: string): ExpoModuleConfig {
+  return new ExpoModuleConfig(require(path) as RawExpoModuleConfig);
```

Comment:
Generally I think we should assume that this file is kind of a partial object. At the end, I don't want people to have to declare all those fields just for common use cases (e.g. I'd like to make `modulesClassNames` optional and when it's not provided automatically find out them). The intention for wrapping it in `ExpoModuleConfig` was that it could provide fallbacks if something is missing in the raw config. But yeah, I agree we should check if provided config is at least type-compatible. I'll add a todo.